code/game/mecha/equipment/tools/medical_tools.dm

```diff
@@ -330,7 +331,7 @@
 			if(M)
 				var/R
 				mechsyringe.visible_message("<span class=\"attack\"> [M] was hit by the syringe!</span>")
-				if(M.can_inject(null, TRUE))
+				if(M.can_inject(originaloccupant, TRUE, original_target_zone))

```

Comment:
Btw, this also makes the occupant see the error message if the body part they are targeting is mechanic/missing/otherwise uninjectable. I'm assuming this was the original intent given the errors=TRUE argument.